nmagent/cachedclient_test.go

```diff
@@ -0,0 +1,79 @@
+package nmagent_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/nmagent"
+	"github.com/google/go-cmp/cmp"
+)
+
+// TestHomeAzCache makes sure the CachedClient works properly in caching home ez response and error
+func TestHomeAzCache(t *testing.T) {
+	tests := []struct {
+		name      string
+		exp       nmagent.HomeAzResponse
+		expPath   string
+		resp      map[string]interface{}
+		shouldErr bool
+	}{
+		{
+			"happy path",
+			nmagent.HomeAzResponse{HomeAz: uint(1)},
+			"/machine/plugins/?comp=nmagent&type=GetHomeAz",
+			map[string]interface{}{
+				"httpStatusCode": "200",
+				"HomeAz":         1,
+			},
+			false,
+		},
+		{
+			"empty response",
+			nmagent.HomeAzResponse{},
+			"/machine/plugins/?comp=nmagent&type=GetHomeAz",
+			map[string]interface{}{
+				"httpStatusCode": "500",
+			},
+			true,
+		},
```

Comment:
These tests make sense for the real client, but for a cache, we want to test something entirely different: does the cache invoke the real client when we expect it to, and does it serve from the cache when we expect it to.